### PR TITLE
Auto-resolve project when project name not specified

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java [hunk 1]
```diff
@@ -132,6 +135,37 @@ protected synchronized void initialize() {
         return completableFuture;
     }
 
+    private IJavaProject calculateJavaProject(String projectName, ThreadReference thread, int depth) {
+        if (StringUtils.isBlank(projectName)) {
```
Comment (akaroml) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java]: Use early return to get rid of long indention block
```java
if (!StringUtils.isBlank(projectName)) {
   return JdtUtils.getJavaProject(projectName);
}
```

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java [hunk 2]
```diff
@@ -70,15 +76,11 @@ public void initialize(IDebugAdapterContext context, Map<String, Object> props)
     @Override
     public CompletableFuture<Value> evaluate(String expression, ThreadReference thread, int depth) {
         CompletableFuture<Value> completableFuture = new CompletableFuture<>();
-        String projectName = (String) options.get(Constants.PROJECTNAME);
+
         if (debugTarget == null) {
+            String projectName = (String) options.get(Constants.PROJECTNAME);
```
Comment (akaroml) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java]: Please put the logic inside the initialize() of this provider. We only need to look for the project once per session.
Trying to calculate project everytime users evaluate expressions is overkill.

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java [hunk 3]
```diff
@@ -132,6 +135,37 @@ protected synchronized void initialize() {
...
+        if (StringUtils.isBlank(projectName)) {
+            try {
+                StackFrame sf = thread.frame(depth);
+                String typeName = sf.location().method().declaringType().name();
+
+                // check whether the project is the only java project in workspace
+                IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+                List<IJavaProject> projects = Arrays.stream(root.getProjects()).map(JdtUtils::getJavaProject).filter(p -> {
+                    try {
+                        return p.findType(typeName) != null && p.hasBuildState();
```
Comment (yaohaizh) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java]: The findType should be heavy API. If there is only one project, we can safely bet the projectName?

**Comment by andxu** [comment]
https://github.com/Microsoft/java-debug/pull/156

**Comment by andxu** [comment]
Since a lot of changes due to conditional bp, create a new pr replacing this one.

**Comment by rajanish-dac** [comment]
Ran into this issue and after few guesses below changing in launch.json helped 

        {
            "type": "java",
            "name": "Attach to Remote Program",
            "request": "attach",
            "hostName": "localhost",
            "port": "5007",
            **"projectName": "foo-bar-service" - this is project name from maven <name>foo-bar-service</name>** 
        }

**Pull request closed without merging**